### PR TITLE
Leave stream open

### PR DESCRIPTION
**Pull request by johnsimons**
Trying to address:
```
An error occurred when sending a request to 'https://52.158.237.122:10933/', after the request began: Object reference not set to an instance of an object.
                    |       Halibut.HalibutClientException
                    |       at Halibut.Transport.SecureListeningClient.HandleError(Exception lastError, Boolean retryAllowed)
                    |       at Halibut.HalibutRuntime.SendOutgoingHttpsRequest(RequestMessage request)
                    |       at Halibut.ServiceModel.HalibutProxy.Invoke(MethodInfo targetMethod, Object[] args)
                    |       at System.Reflection.DispatchProxyGenerator.Invoke(Object[] args)
                    |       at generatedProxy_1.GetStatus(ScriptStatusRequest )
                    |       at Octopus.Server.Orchestration.Targets.Tentacles.ScriptObserver.ObserveUntilComplete(ScriptTicket ticket, Int64 next, ScriptOutputFilter output) in C:\buildAgent\work\abb2fbfce959a439\source\Octopus.Server\Orchestration\Targets\Tentacles\ScriptObserver.cs:line 45
                    |       at Octopus.Server.Orchestration.Targets.Tentacles.ScriptObserver.ObserveThenFinish(ScriptTicket ticket, ScriptOutputFilter output, Action`1 collectArtifacts) in C:\buildAgent\work\abb2fbfce959a439\source\Octopus.Server\Orchestration\Targets\Tentacles\ScriptObserver.cs:line 34
                    |       at Octopus.Server.Orchestration.Targets.Tentacles.ScriptObserver.ObserveUntilComplete(ScriptTicket ticket, Action`1 collectArtifacts) in C:\buildAgent\work\abb2fbfce959a439\source\Octopus.Server\Orchestration\Targets\Tentacles\ScriptObserver.cs:line 27
                    |       at Octopus.Server.Orchestration.ServerTasks.Deploy.ActionExecution.Immediate.ExecutionTargets.TentacleExecutionTarget.Execute(ScriptCollection bootstrapperScripts, IReadOnlyList`1 boostrapperArguments, IReadOnlyList`1 files, Nullable`1 forceIsolationLevel, Boolean raw, Nullable`1 isolationMutexTimeout) in C:\buildAgent\work\abb2fbfce959a439\source\Octopus.Server\Orchestration\ServerTasks\Deploy\ActionExecution\Immediate\ExecutionTargets\TentacleExecutionTarget.cs:line 72
                    |       at Octopus.Server.Orchestration.ServerTasks.Deploy.ActionExecution.Immediate.ImmediateExecutor.ExecuteCalamari(ICalamariFlavour flavour, String calamariCommand, IReadOnlyList`1 calamariArguments, IReadOnlyList`1 files, IReadOnlyList`1 deploymentTools, VariableCollection extraVariables, TargetManifest targetManifest, Nullable`1 isolationMutexTimeout) in 
                    |       --Inner Exception--
                    |       Object reference not set to an instance of an object.
                    |       System.NullReferenceException
                    |       at Halibut.Transport.Protocol.MessageExchangeStream.ReadBsonMessage[T]()
                    |       at Halibut.Transport.Protocol.MessageExchangeStream.Receive[T]()
                    |       at Halibut.HalibutRuntime.<>c__DisplayClass34_0.<SendOutgoingHttpsRequest>b__0(MessageExchangeProtocol protocol)
                    |       at Halibut.Transport.SecureListeningClient.ExecuteTransaction(Action`1 protocolHandler)
```

My theory is that we are somehow closing the stream, so in this PR I used the overloads to ensure the `leaveOpen` flag is set to `true`

### CODE REVIEW

File: source/Halibut/Transport/Protocol/MessageExchangeStream.cs [hunk 1]
```diff
@@ -315,16 +321,18 @@ void WriteEachStream(IEnumerable<DataStream> streams)
...
+                using (var writer = new BinaryWriter(stream, Utf8EncodingNoMarkerThrow, true))
+                {
+                    writer.Write(dataStream.Id.ToByteArray());
+                    writer.Write(dataStream.Length);
+                    writer.Flush();
 
-                ((IDataStreamInternal)dataStream).Transmit(stream);
-                stream.Flush();
+                    ((IDataStreamInternal) dataStream).Transmit(stream);
+                    stream.Flush();
```
Comment (benPearce1) [inline source/Halibut/Transport/Protocol/MessageExchangeStream.cs]: formatting?

**Comment by johnsimons** [comment]
> I am not convinced by this change. I do have concerns that it might be unnecessary and the problem this is trying to address is in the worker services not halibut

@benPearce1 I am not convinced either, I am going by what the stacktrace is telling us, as far as I can see the stacktrace seems to point to a bug in Halibut

**Comment by johnsimons** [comment]
I'll close this one for now

**Comment by jzabroski** [comment]
Can you use MbDotNet to test it?

**Pull request closed without merging**